lib/assets/data_frame_cell/main.js

```diff
@@ -0,0 +1,571 @@
+              />
+            </svg>
+          </button>
+        </span>
+      </div>
+    </div>
+  `,
+  };
+
+  const BaseButton = {
+    name: "BaseButton",
+    props: {
+      label: {
+        type: String,
+        default: "",
+      },
+    },
+    template: `
+    <button class="button button--sm button--dashed" type="button" :disabled="noDataFrame"
+      @click="$emit('addOperation')">
+      <svg width="10" height="10" viewBox="0 0 10 10" fill="currentColor" xmlns="http://www.w3.org/2000/svg">
+        <path d="M4.41699 4.41602V0.916016H5.58366V4.41602H9.08366V5.58268H5.58366V9.08268H4.41699V5.58268H0.916992V4.41602H4.41699Z"/>
+      </svg>
+      {{ label }}
+    </button>
+    `,
+  };
+
+  const BaseInputIcon = {
+    name: "BaseInputIcon",
+    props: {
+      label: {
+        type: String,
+        default: "",
+      },
+    },
+    template: `
+    <div class="icon-container" @click="$emit('removeOperation')">
+      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" width="16" height="16">
+        <path fill="none" d="M0 0h24v24H0z"/>
+        <path d="M17 6h5v2h-2v13a1 1 0 0 1-1 1H5a1 1 0 0 1-1-1V8H2V6h5V3a1 1 0 0 1 1-1h8a1 1 0 0 1 1 1v3zm1 2H6v12h12V8zm-9 3h2v6H9v-6zm4 0h2v6h-2v-6zM9 4v2h6V4H9z"/>
+      </svg>
+    </div>
+    `,
+  };
+
+  const app = Vue.createApp({
+    components: { BaseSelect, BaseInput, Accordion, BaseButton, BaseInputIcon },
+    template: `
+      <div class="app">
+        <!-- Info Messages -->
+        <div class="box box-warning" v-if="missingRequire">
+          <p>To successfully work with Data Transform smart cells, you need to add the following require:</p>
+          <pre><code>{{ missingRequire }}</code></pre>
+        </div>
+        <div class="box box-warning" v-if="noDataFrame">
+          <p>The Data Transform smart cells works with Explorer DataFrames but none was found.</p>
+          <p>To get started quickly, you might run the following on a previous cell:</p>
+          <pre><code>iris = Explorer.Datasets.iris()</code></pre>
+        </div>
+        <!-- Data Frame Form -->
+        <form @change="handleFieldChange">
+          <div class="container">
+            <div class="root">
+              <BaseSelect
+                name="data_frame"
+                label="Data Frame"
+                v-model="rootFields.data_frame"
+                :options="dataFrameVariables"
+                :required
+                :disabled="noDataFrame"
+                class="root-field"
+              />
+              <BaseInput
+                name="export_to"
+                label="Export to"
+                v-model="rootFields.export_to"
+                :disabled="noDataFrame"
+                class="root-field"
+              />
+            </div>
+            <div class="operations" v-if="hasOperations('sorting')">
+              <Accordion
+                class="operations-wrapper"
+                v-for="(sort, index) in operations.sorting"
```

Comment:
We currently group operations by type, but we should really allow adding them in arbitrary order, because otherwise it limits the transformations. For example the user may want to pivot, then filter; and it may become more relevant as we start adding more operations.

Sidenote: usually it would make sense to put sorting at the end, but if we leave the order up to the user, it's no longer a concern on our side.

Reply:
Yep! We will allow arbitrary order soon, but for now let's put sorting at the end.